packages/perseus/src/validation.types.ts

```diff
@@ -95,9 +97,38 @@ export type PerseusInputNumberUserInput = {
     currentValue: string;
 };
 
+// Would it be better just to use a question mark instead of having all of them also have undefined?
+// Are those two different types though? optional versus required with an undefined value?
+export type PerseusGraphCorrectType = {
```

Comment:
I'd recommend using a union type here where each `type` is associated with the actual set of data that it requires.  

Eg: 
```ts
export type PerseusGraphCorrectType = {
    type: "angle";
    coords: [Coord, Coord, Coord];
    allowReflexAngles: boolean | undefined;
} | {
    type: "circle";
    ... whatever circle needs
} | ... other types. 
```

Reply:
This is exactly what I was working on before, so I'll just go back to that version then :) Thanks!